### DISCUSSION TITLE
Correct colors on Brother Machine

### DISCUSSION THREAD

**Discussion by CheerfulTomato** [inkstitch/inkstitch#2595] [Q&A]
Hi,

I followed [this tutorial](https://inkstitch.org/de/docs/thread-color/) to create my own palette for Madeira Cotton 30. From inkscape perspective it looks successful, I can select the colors and if I hover over them, I see name + number.

When exporting to PES (my machine accepts PES, PEN, PHC and DST, so PES seems to be my only option) and starting the embroidery, my machine shows other colors (probably mapped somehow). I saw [in some issues](https://github.com/inkstitch/inkstitch/issues/1105), that PES seems to be limited in what colors could be used, but cannot find this information [in the documentation about PES](https://edutechwiki.unige.ch/en/Embroidery_format_PES#Color_subsection).

So, am I doing something wrong? I imported my palette to inkscape only (by moving the file to the inkscape folder), not to inkstitch, is there an additional step needed?

As a workaround I would also be happy if it would show either the correct name or correct number. It's okay if the shown color is a bit off.

Also, my machine shows which step is coming up, so the PDF export allows me to pick the correct color, so this is more a quality of life question rather than a showstopper. I just digitized my first own design after owning a machine for two days. I must say, I love it! Such a good result for a first, so thank you very much for providing such an amazing piece of software :-)

Kind regards!

**Comment by tyrosinase** [comment]
Machines seem to, yeah, do their own mappings of what's in the file. Brothers, in particular, have some dedicated stitches that they will display if you select certain Sulky pinks (they'll show a trim or whatever, though they'll go ahead and stitch normally). It would be nice if we could back-calculate what it takes to get them to land on the correct color, and create a palette specific to different machines.

I thiiiiink we could do that by creating some files that just have a bunch of RGB combinations, and people could load them up and report back on what colors their machines assign to each one. Tedious, but not technically difficult.

**Comment by CheerfulTomato** [comment]
Interestingly, the colors seem to be already off in the PES. The Bernina V9 Creator software shows the exact same colors as my machine (which are quite far off from what I set in inkscape/-stitch). There's even one color that is being matched to "Position of Appliquée" (or something alike)... So can it be, that this is an issue of inkstitch or my palette setup?

**Comment by CheerfulTomato** [comment]
Also interesting: If I modify the color of thread in Bernina V9, the name of the machine is still altered, but at least the color number is correct. So I assume this is a bug in inkstitch, filing a bug now. Maybe someone finds the issue :-)

**Reply by CheerfulTomato** [reply]
created issue: https://github.com/inkstitch/inkstitch/issues/2612
